core/jvm/src/main/scala/zio/sql/expr.scala

```diff
@@ -210,6 +226,10 @@ trait ExprModule extends NewtypesModule with FeaturesModule with OpsModule {
 
   sealed case class FunctionDef[-A, +B](name: FunctionName) { self =>
 
+//    def apply[Source, B1 >: B]()(implicit typeTag: TypeTag[B1]): Expr[Unit, Source, B1] =
+    def apply[B1 >: B]()(implicit typeTag: TypeTag[B1]): Expr[Features.Function0, Any, B1] =
```

Comment:
can we add implicit `ev :  Any <:< A`? (see other apply methods). This will make sure only a FunctionCall0 can call apply with no arguments.